news/templates/news/user_news.html

```diff
@@ -1,47 +1,51 @@
-  </h4>
-  <div>
-    {{ item.body|safe }}
-  </div>
-</div>
-{% endfor %}
 
 {% if news %}
-<div class="panel panel-default news-panel">
-  <div class="panel-heading">
-    <h4 class="panel-title">{% trans "Course news" %}</h4>
-  </div>
-  <div class="list-group">
-    {% for item in news %}
-    <a class="list-group-item{% if item.collapsed %} collapsed{% endif %}" href="#news{{ item.id }}" data-toggle="collapse" role="button" aria-controls="news{{ item.id }}" aria-expanded="{% if item.collapsed %}false{% else %}true{% endif %}">
-      <h4 class="list-group-item-heading">
-        <span class="caret"></span>
-        {{ item.title|safe }} <small>{{ item.publish }}</small>
-        {% if not item|is_published:now %}<small><span class="label label-default">{% trans "Appears later" %}</span></small>{% endif %}
-        {% if is_course_staff %}<small><span class="label label-default">{{ item.audience|news_audience }}</span></small>{% endif %}
-      </h4>
-    </a>
-    <div class="list-group-item-text collapse{% if not item.collapsed %} in{% endif %}" id="news{{ item.id }}">
-      {{ item.body|safe }}
-    </div>
-    {% if item.begin_more %}
-    <a class="list-group-item" href="#more-news" data-toggle="collapse" role="button" aria-controls="more-news" aria-expanded="false">
-      {% trans "Show more" %}
-    </a>
-    <div class="collapse" id="more-news">
-    {% endif %}
-    {% endfor %}
-    {% if more > 0 and news|length > more %}
-    </div>
-    {% endif %}
-  </div>
-</div>
+	<div class="panel panel-default news-panel">
+		<div class="panel-heading">
+			<h3 class="panel-title">{% trans "Course news" %}</h3>
+		</div>
+		<div class="list-group">
+			{% for item in news %}
+				<div class="{% if item.pin %}pinned-{% endif %}list-group-item">
+					{% if item.pin %}
+						<span class="sr-only">{% trans "Pinned news item on top of the list" %}</span>
+					{% endif %}
+					<div class="list-group-item-heading">
+						<h4 class="list-group-item-title">
+							{{ item.title|safe }}
+						</h4>
+						<div class="list-group-item-details">
+							{% if not item|is_published:now %}
+								<span class="future-instance">{% trans "Published on" %}: <time>{{ item.publish }}</time></span>
+							{% else %}
+								<time class="current-instance">{{ item.publish }}</time>
+							{% endif %}
+							{% if is_course_staff %}
+								<span class="label label-primary">{{ item.audience|news_audience }}</span>
+							{% endif %}
+							{% if item.pin %}
+								<span class="glyphicon glyphicon-pushpin" aria-hidden="true"></span>
+							{% endif %}
+						</div>
+					</div>
+					<p class="list-group-item-text">
+						{{ item.body|safe }}
+					</p>
+				</div>
+				{# If a limit for folding news (more) is set, this creates the folding section #}
+				{% if forloop.counter == more and news|length > more and more > 0 %}
```

Comment:
I think this code shows 4 news items when more is set to 3. I think this element should be before the news item. Thus, when index (`forloop.counter`) is 3, then there has been indexes 0, 1, and 2 (i.e. 3 news items). Technically it doesn't matter so much, as more can be selected correctly. Though, I didn't yet test this and I'm not sure how moving this works with the `news|lenght > more` test.

Reply:
Hmm.. I am not quite sure what you mean? Currently the more is set to 3 in the course.html (that has not been changed) and above the folding three items are shown.

So in that sense the limit for folding works like before, just without the for-loop in the news.

Reply:
I see.. I always forget that `forloop.counter` starts from 1, thus this works correctly. Alternatively, the test could be before the news item and be `forloop.counter0 == more and more > 0` (no need to check length of the news as that would be the first entry whcih goes within the collapsed section. Current solution is totally fine :)

Reply:
:D